sample/sample.go

```diff
@@ -64,6 +145,24 @@ func Importance(samples, weights []float64, target dist.LogProber, proposal dist
 	}
 }
 
+// Rejectioner is a wrapper around the Rejection sampling generation procedure.
+// If the rejection sampling fails during the call to Sample, all samples will
+// be set to math.NaN().
+type Rejectioner struct {
```

Comment:
I'm not suggesting removing the current behaviour, just adding the error field. Maybe with a method so it can be generically type asserted by the caller.